### PR TITLE
feat: Set up stream chat api and sign up api route

### PR DESCRIPTION
**Pull request by BhanuPratap-Singh-Naruka**
### Task Link

https://www.notion.so/BE-Signup-API-StreamChat-setup-532ff5907a974b2fb79d1e1d432d83a3?pvs=4

### Description

* Set up signup route API  
* For messaging functionality set up stream-chat

#### Sample Curl :-

```

curl --location 'http://localhost:2999/signup' \
--header 'Content-Type: application/json' \
--data '{
    "id": "11",
    "name": "John Doe"
}'

```

#### Sample response :-
 
```
{
    "id": "11",
    "role": "user",
    "created_at": "2023-03-13T03:50:04.152025Z",
    "updated_at": "2023-03-13T03:50:04.152025Z",
    "banned": false,
    "online": false,
    "name": "John Doe"
}
```

### TODO Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

File: server/.env [hunk 1]
```diff
@@ -1 +1,4 @@
-PORT = 2999
\ No newline at end of file
+PORT = 2999
+CLIENT_URL = http://127.0.0.1:5173/
```
Comment (abhimishra01) [inline server/.env:2]: `CLIENT_URL = http://localhost:5173/`

File: server/.env [hunk 2]
```diff
@@ -1 +1,4 @@
...
+STREAM_API_KEY= efb7xtpvjqsx
```
Comment (abhimishra01) [inline server/.env:3]: well, this is not a good practise but for now lets keep it

we will use [github secrets](https://github.com/Azure/actions-workflow-samples/blob/master/assets/create-secrets-for-GitHub-workflows.md) when our CD pipeline is ready

File: server/server.ts [hunk 3]
```diff
@@ -2,7 +2,11 @@ import { config } from "dotenv";
 config();
 
 import fastify from "fastify";
+import cors from "@fastify/cors";
+import { userRoutes } from "./routes/users";
```
Comment (abhimishra01) [inline server/server.ts:6]: lets have a space and try to use alias based imports

**Review by abhimishra01 — CHANGES_REQUESTED** [review]

File: server/server.ts (hunk 3 above)
Comment (BhanuPratap-Singh-Naruka) [reply server/server.ts:6]: I have tried alias based import for routes folder but for some reason it's throwing error and crashing the server.

File: server/.env (hunk 2 above)
Comment (BhanuPratap-Singh-Naruka) [reply server/.env:3]: okay. That would be good

**Comment by abhimishra01** [comment]
please resolve merge conflicts

**Comment by abhimishra01** [comment]
Also, please add a postman curl and sample Response for the API

**Review by abhimishra01 — APPROVED** [review]

**Review by saloni24gupta — APPROVED** [review]

**Review by saloni24gupta — APPROVED** [review]

**Review by saloni24gupta — APPROVED** [review]

**Pull request merged**